Crashlytics/UnitTests/FIRCLSRecordTests.m

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+
+#import "FIRCLSRecordAdapter.h"
+#import "FIRCLSRecordAdapter_Private.h"
+#import "FIRCLSRecordApplication.h"
+#import "FIRCLSRecordBase.h"
+#import "FIRCLSRecordBinaryImage.h"
+#import "FIRCLSRecordExecutable.h"
+#import "FIRCLSRecordFrame.h"
+#import "FIRCLSRecordHost.h"
+#import "FIRCLSRecordIdentity.h"
+#import "FIRCLSRecordKeyValue.h"
+#import "FIRCLSRecordProcessStats.h"
+#import "FIRCLSRecordRegister.h"
+#import "FIRCLSRecordRuntime.h"
+#import "FIRCLSRecordSignal.h"
+#import "FIRCLSRecordStorage.h"
+#import "FIRCLSRecordThread.h"
+
+@interface FIRCLSRecordTests : XCTestCase
+
+@end
+
+@implementation FIRCLSRecordTests
+
+/// It is important that crashes do not occur when reading persisted crash files before uploading
+/// Verify various invalid input cases
+- (void)testInvalidRecordCases {
+  id adapter __unused = [[FIRCLSRecordAdapter alloc] initWithPath:@"nonExistentPath"];
```

Comment:
Could we add a test case where we initialize with a valid path, but files in the path are corrupted / invalid JSON?

Reply:
Added.